### PR TITLE
Fix closeable resource leaks in Junit tests

### PR DESCRIPTION
**Pull request by lilgreenbird**
1st part of Junit test cleanup:
-  use try-with-resources on closeables and fix leaks
- get rid shared variables between tests
- move testframework/util/Util.java to TestUtil (part of work get rid of testframework)

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/connection/PoolingTest.java [hunk 1]
```diff
@@ -81,15 +82,32 @@ public void testPooling() throws SQLException {
...
+                }
+                if (null != con2) {
+                    con2.close();
+                }
+            }
+        } finally {
+            if (null != pc) {
+                pc.close();
+            }
+        }
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/connection/PoolingTest.java]: This test can be compressed as well.. with try-with-resources.

File: src/test/java/com/microsoft/sqlserver/jdbc/connection/TimeoutTest.java [hunk 2]
```diff
@@ -193,39 +224,59 @@ public void testCancelQueryTimeoutOnStatement() throws Exception {
...
+                throw new Exception(TestResource.getResource("R_expectedExceptionNotThrown"));
+            } catch (Exception e) {
+                if (!(e instanceof SQLException)) {
+                    throw e;
+                }
+                assertEquals(e.getMessage(), TestResource.getResource("R_readTimedOut"),
+                        TestResource.getResource("R_invalidExceptionMessage"));
+            }
+            try {
+                conn.createStatement().execute("SELECT @@version");
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/connection/TimeoutTest.java]: Hidden statement here.. use the same 'stmt' instance instead

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataForeignKeyTest.java [hunk 3]
```diff
@@ -100,22 +95,40 @@ public static void terminateVariation() throws SQLException {
...
-        } catch (SQLException e) {
-            assertTrue(e.getMessage().startsWith(TestResource.getResource("R_dbNameIsCurrentDB")));
+        SQLServerResultSet rs1 = null;
+        SQLServerResultSet rs2 = null;
+        SQLServerResultSet rs3 = null;
+     
+        try (SQLServerConnection conn = (SQLServerConnection) DriverManager.getConnection(connectionString)) {
+            SQLServerDatabaseMetaData dmd = (SQLServerDatabaseMetaData) conn.getMetaData();
+
+            rs1 = (SQLServerResultSet) dmd.getImportedKeys(null, null, table1);
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataForeignKeyTest.java]: Move all ResultSets to one try w. res. block - validate method calls can happen all at once inside the block..

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java [hunk 4]
```diff
@@ -1866,7 +1866,7 @@ public void testNBCROWWithRandomAccess() throws Exception {
                 con = ds.getConnection();
                 Statement stmt = con.createStatement();
```
Comment (cheenamalhotra) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/StatementTest.java]: Has this class been worked upon too? I see lot of connection/statements not closed properly.
Also `Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");` is unnecessary inside tests.. let's remove them too.

File: src/test/java/com/microsoft/sqlserver/jdbc/TestUtils.java [hunk 5]
```diff
@@ -203,15 +566,15 @@ public static Object roundDatetimeValue(Object value) {
             case 1:
                 ts.setNanos((millis - 1) * 1000000);
                 break;
-
+
```
Comment (rene-ye) [inline src/test/java/com/microsoft/sqlserver/jdbc/TestUtils.java]: Some weird spacing issues here.

**Pull request merged**